memgraph-graphRAG/graphRAG.ipynb

```diff
@@ -0,0 +1,797 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# GraphRAG in Memgraph\n",
+    "\n",
+    "In this tutorial, we will build GraphRAG using the Memgraph ecosystem and\n",
+    "OpenAI. This example is based on a portion of a fixed Game of Thrones dataset,\n",
+    "which will be enriched with unstructured data to create a knowledge graph. \n",
+    "\n",
+    "To search for relevant information, in this example we will use vector search on\n",
+    "node embeddings to find schematically relevant data. Following this, the\n",
+    "structured data will be extracted from the graph and passed to LLM to answer the\n",
+    "question. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "To begin with this tutorial, you will need Docker, Python and an OpenAI API key.\n",
+    "With a few small tweaks, you can adapt this setup to run on your local Ollama\n",
+    "environment. \n",
+    "\n",
+    "First, we need to start Memgraph with the vector search capabilities. You can do\n",
+    "this by running the following command: \n",
+    "\n",
+    "TODO: Updated the command when the vector search is available in the official\n",
+    "Memgraph docker image\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "\n",
+    "docker run -p 7687:7687 -p 7444:7444 memgraph/memgraph-mage:exp-vector-1 --log-level=TRACE --also-log-to-stderr --telemetry-enabled=False --experimental-vector-indexes='tag__Entity__embedding__{\"dimension\":384,\"limit\":3000}'\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "You can run this command outside of this notebook. \n",
+    "\n",
+    "Once Memgraph is running in the background, make sure to load the initial Game\n",
+    "of Thrones dataset:  "
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
If a command is not intended to be run inside this notebook then it doesn't go into the code cell type. Put it into the markdown cell type and add backticks for code rendering.